File: tournaments/comfortableNumbers/comfortableNumbers.cs
int comfortableNumbers(int l, int r) {
    var result = 0;

    for (var i = l; r >= i; ++i) {
        var iSum = getSumOfDigits(i);
        for (var j = i + 1; Math.Min(i+iSum,r) >= j; ++j) {
            var jSum = getSumOfDigits(j);

            if (i >= j - jSum) {
                ++result;
            }
        }
    }

    return result;
}

int getSumOfDigits(int n) {
    var result = 0;

    while (0 < n) {
        result += n % 10;
        n /= 10;
    }

    return result;
}

File: tournaments/comfortableNumbers/comfortableNumbers.js
function comfortableNumbers(l, r) {
    const getSumOfDigits = function(n) {
        let r = 0;
        while (n) {
            r += n % 10;
            n = ~~(n / 10);
        }
        return r;
    };

    let result = 0;

    for (let i = l; r >= i; ++i) {
        const a = getSumOfDigits(i);
        for (let j = i + 1; Math.min(i + a, r) >= j; ++j) {
            const b = getSumOfDigits(j);

            if (i >= j - b)
                ++result;
        }
    }

    return result;
}
